packages/compartment-mapper/src/bundle-lite.js

```diff
@@ -0,0 +1,612 @@
+ * variable.
+ * @property {() => string} getCells Produces a JavaScript string consisting of
+ * a JavaScript object and a trailing comma.
+ * The string is evaluated in a lexical context with a `cell` maker, the `cells`
+ * array of every module's internal environment record.
+ * @property {() => string} getFunctorCall Produces a JavaScript string may
+ * be a statement that calls this module's functor with the calling convention
+ * appropriate for its language, injecting whatever cells it needs to link to
+ * other module namespaces.
+ * @property {() => string} getReexportsWiring Produces a JavaScript string
+ * that may include statements that bind the cells reexported by this module.
+ */
+
+/**
+ * @template {unknown} SpecificModuleSource
+ * @typedef {object} BundleModule
+ * @property {string} key
+ * @property {string} exit
+ * @property {string} compartmentName
+ * @property {string} moduleSpecifier
+ * @property {string} sourceDirname
+ * @property {string} parser
+ * @property {StaticModuleType & SpecificModuleSource} record
+ * @property {Record<string, string>} resolvedImports
+ * @property {Record<string, number>} indexedImports
+ * @property {Uint8Array} bytes
+ * @property {number} index
+ * @property {BundlerKit} bundlerKit
+ */
+
+/**
+ * @typedef {object} BundleExit
+ * @property {string} exit
+ * @property {number} index
+ * @property {BundlerKit} bundlerKit
+ * @property {Record<string, number>} indexedImports
+ * @property {Record<string, string>} resolvedImports
+ */
+
+/**
+ * @template {unknown} SpecificModuleSource
+ * @callback GetBundlerKit
+ * @param {BundleModule<SpecificModuleSource>} module
+ * @param {object} params
+ * @param {boolean} [params.useEvaluate]
+ * @param {string} [params.sourceUrlPrefix]
+ * @returns {BundlerKit}
+ */
+
+/**
+ * @template {unknown} SpecificModuleSource
+ * @typedef {object} BundlerSupport
+ * @property {string} runtime
+ * @property {GetBundlerKit<SpecificModuleSource>} getBundlerKit
+ */
+
+import { resolve } from './node-module-specifier.js';
+import { link } from './link.js';
+import { makeImportHookMaker } from './import-hook.js';
+import { defaultParserForLanguage } from './archive-parsers.js';
+
+import mjsSupport from './bundle-mjs.js';
+import cjsSupport from './bundle-cjs.js';
+import jsonSupport from './bundle-json.js';
+
+const { quote: q } = assert;
+
+/**
+ * @param {BundleExit} source
+ * @returns {BundlerKit}
+ */
+const makeCjsExitBundlerKit = ({ exit }) => ({
+  getFunctor: () => `\
+null,
+`,
+  getCells: () => `\
+    namespaceCells(tryRequire(${JSON.stringify(exit)})),
+`,
+  getReexportsWiring: () => '',
+  getFunctorCall: () => ``,
+});
+
+/**
+ * @param {Record<string, CompartmentDescriptor>} compartmentDescriptors
+ * @param {Record<string, CompartmentSources>} compartmentSources
+ * @param {string} entryCompartmentName
+ * @param {string} entryModuleSpecifier
+ * @param {Array<string>} exitModuleSpecifiers
+ */
+const sortedModules = (
```

Comment:
Can you add a description for `sortedModules`? If I'm reading correctly, it's basically a bottom-up topological traversal of the module graph (i.e., a generalization of post-order [tree traversal](https://en.wikipedia.org/wiki/Tree_traversal#Depth-first_search) to handle non-tree graphs) with a side table at each node for alias lookup?

Reply:
Got it in one.